### PR TITLE
R3 - add features to index

### PR DESCRIPTION
**Pull request by SethDamiani**
## Summary of PR

Implemented all features of R3, including:

- Add user balance column to User model, show on index page
- Add forms for adding, editing, and buying tickets to the index page

### CODE REVIEW

File: qa327/frontend.py [hunk 1]
```diff
@@ -109,6 +109,8 @@ def wrapped_inner():
                 # if the user exists, call the inner_function
                 # with user as parameter
                 return inner_function(user)
+            else:
+                return redirect('/login')
         else:
             # else, redirect to the login page
             return redirect('/login')
```
Comment (atangent) [inline qa327/frontend.py]: ## Don't think the `else` statements and extra `redirect` are necessary. You can achieve the same with
```
if 'logged_in' in session:
  # assign variables

  if user:
    return inner_function(user)

return redirect('/login')
```
If we are logged in and there is no user, we would never enter the `if user` condition. We would correctly proceed to `redirect` to login.

If we're not logged in, we wouldn't enter the `if logged_in` condition. Likewise, we correctly proceed to `redirect` as well.

### On considering the side effects of taking out `else`.
But would we `return redirect` by accident if there are no `else` statements? Thankfully, we can't. 

The only time we don't `return redirect` is when we `return inner_function` instead. This is taken care of by the one `if user` condition. 

Therefore, taking out the `else`s would not mess up the logic.

### The one `redirect` line does the job and allows for cleaner code.

**Review by atangent — COMMENTED** [review]
Getting rid of the `else` statements and extra `redirect` makes the code more readable

**Comment by atangent** [comment]
Looks good-to-merge to me now

**Review by atangent — APPROVED** [review]

**Review by Kymed — COMMENTED** [review]
Interesting choice to put line breaks after each form inputs. Looks good to me!

**Pull request merged**